core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-theme-selector/template-builder-theme-selector.component.ts

```diff
@@ -0,0 +1,179 @@
+import { fromEvent as observableFromEvent, Subject } from 'rxjs';
+
+import { CommonModule } from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    EventEmitter,
+    Input,
+    OnDestroy,
+    OnInit,
+    Output,
+    ViewChild
+} from '@angular/core';
+
+import { LazyLoadEvent, MessageService } from 'primeng/api';
+import { ButtonModule } from 'primeng/button';
+import { DataView, DataViewModule } from 'primeng/dataview';
+import { DropdownModule } from 'primeng/dropdown';
+import { DialogService, DynamicDialogConfig, DynamicDialogRef } from 'primeng/dynamicdialog';
+
+import { debounceTime, take, takeUntil } from 'rxjs/operators';
+
+import { PaginatorService } from '@dotcms/data-access';
+import { Site, SiteService } from '@dotcms/dotcms-js';
+import { DotTheme } from '@dotcms/dotcms-models';
+import { DotMessagePipeModule, DotSiteSelectorDirective } from '@dotcms/ui';
+
+/**
+ * The DotThemeSelectorComponent is modal that
+ * show the themes of the available hosts
+ * @export
+ * @class DotThemeSelectorComponent
+ */
+@Component({
+    selector: 'dotcms-template-builder-theme-selector',
+    providers: [PaginatorService, DialogService, MessageService],
+    imports: [
+        ButtonModule,
+        DropdownModule,
+        DotSiteSelectorDirective,
+        DataViewModule,
+        CommonModule,
+        DotMessagePipeModule
+    ],
+    standalone: true,
+    templateUrl: './template-builder-theme-selector.component.html',
+    styleUrls: ['./template-builder-theme-selector.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TemplateBuilderThemeSelectorComponent implements OnInit, OnDestroy {
+    ref: DynamicDialogRef;
+    themes: DotTheme[] = [];
+
+    @Input()
+    value: DotTheme;
+
+    @Output()
+    selected = new EventEmitter<DotTheme>();
+
+    @ViewChild('searchInput', { static: true })
+    searchInput: ElementRef;
+
+    @ViewChild('dataView', { static: true })
+    dataView: DataView;
+
+    @ViewChild('siteSelector', { static: true })
+    current: DotTheme;
+    visible = true;
+
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+    private SEARCH_PARAM = 'searchParam';
+    private initialLoad = true;
+
+    constructor(
+        public dialogService: DialogService,
+        public messageService: MessageService,
+        public paginatorService: PaginatorService,
+        private siteService: SiteService,
+        public cd: ChangeDetectorRef,
+        private config: DynamicDialogConfig
+    ) {}
+
+    ngOnInit() {
+        this.current = this.value;
+        this.paginatorService.url = 'v1/themes';
```

Comment:
Maybe this can be a constant

Reply:
Not need, is how the paginator service work.